### PR TITLE
 resources: aws: ec2: CheckApply fix pending and stopping cases

### PR DESCRIPTION
**Pull request by jonathangold**
Found this bug while trying to fix the other one. :P

### CODE REVIEW

File: resources/aws_ec2.go [hunk 1]
```diff
@@ -785,6 +785,13 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 	ctx, cancel := context.WithTimeout(context.TODO(), waitTimeout*time.Second)
 	defer cancel()
 	if len(diOutput.Reservations) == 1 {
+		// If the instance is in the process of starting up or stopping
+		// we cannot affect its state until it finishes and Watch sends
+		// an event.
+		initialState := *diOutput.Reservations[0].Instances[0].State.Name
```
Comment (purpleidea) [inline resources/aws_ec2.go]: Is it possible that it gets into this state *right* after we check this?

Comment (jonathangold) [reply resources/aws_ec2.go]: @purpleidea the short answer is yes. 
Instead I should be checking for the error that's produced where I try to start/stop/terminate the instances and return false, nil there.

File: resources/aws_ec2.go [hunk 2]
```diff
@@ -798,6 +798,13 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 			}
 			_, err := obj.client.StartInstances(startInput)
 			if err != nil {
+				// If the instance is not in a state where it
+				// can be started, we can't do anything.
+				if aerr, ok := err.(awserr.Error); ok {
+					if aerr.Code() == "IncorrectInstanceState" {
```
Comment (purpleidea) [inline resources/aws_ec2.go]: If we're going to have all these absurd strings everywhere let's at least store them as `const` Maybe something like `AwsErrWhatever`...

**Comment by jonathangold** [comment]
@purpleidea fixed. :)

**Pull request merged**